src/core/control/jobs/Job.h

```diff
@@ -11,15 +11,24 @@
 
 #pragma once
 
-#include <mutex>
-#include <string>
-#include <vector>
+#include <atomic>
 
 enum JobType { JOB_TYPE_BLOCKING, JOB_TYPE_PREVIEW, JOB_TYPE_RENDER, JOB_TYPE_AUTOSAVE };
 
+/**
+ * A manually ref-counted class representing an asynchronous job to be used with
+ * the Xournal++ scheduler.
+ *
+ * Instances of Job (and its subclasses) should be constructed exclusively using
+ * the `new` keyword.
+ */
```

Comment:
Maybe it would be better to use `std::shared_ptr` or even std::shared_from_this (depends, if this is passed to C functions, and if this is done in an async OP; then maybe not). 

Also, a `JopPtr create(Args... args)` and a private/protected constructor will prevent the usage of stack allocations.

Reply:
We're not going to pull in an extra dependency for a simple refactor.

Reply:
I read over the docs for `std::shared_ptr` again and found that its ref count is actually handled with atomics, so it would be better to use `std::shared_ptr`. But that would involve quite a bit more effort.